src/main/java/gui/EditRulePopup.java

```diff
@@ -240,17 +282,26 @@ private JPanel createEditorComplexityTogglePanel() {
 		basicComplexity.setEnabled(false);
 
 		basicComplexity.addActionListener(e -> {
-			advancedComplexity.setEnabled(true);
-			advancedMode = false;
-			createMetricsPanel();
-			metricsPanel.revalidate();
-			metricsPanel.repaint();
-			basicComplexity.setEnabled(false);
+			if (rule.isAdvanced() || rule.isDefault() || rule.getName() != "") {

```

Comment:
This way you don't need the else.
```suggestion
			if (!(rule.isAdvanced() || rule.isDefault() || rule.getName() != "")) {
			advancedComplexity.setEnabled(true);
 			basicComplexity.setEnabled(false);		 				advancedMode = false;
 				createMetricsPanel();
 				metricsPanel.revalidate();
 				metricsPanel.repaint();
 				basicComplexity.setEnabled(false);
 			}
```